Spec/RealtimeClientConnection.swift

```diff
@@ -3028,26 +2989,165 @@ class RealtimeClientConnection: QuickSpec {
                     default:
                         fail("unhandled: \(fixtureMessage["expectedType"].string!)")
                     }
+                }
 
-                    waitUntil(timeout: testTimeout) { done in
-                        channel.publish([message]) { err in
-                            if let err = err {
-                                fail("\(err)")
+                // https://github.com/ably/wiki/issues/22
+                it("should encode and decode fixture messages as expected") {
+                    let options = AblyTests.commonAppSetup()
```

Comment:
Hmm, I don't see why. Whether this uses JSON or MsgPack doesn't matter, that's not the layer we're testing. We have fully independent two layers:

* Message data encoding JSON-like to string and back again.
* Whole protocol message encoding to JSON or MsgPack and back again.

This test is about the former. When the encoded message data hits the latter, it is already a string in all cases.